web_translate_dialog/static/src/js/web_translate_dialog.js

```diff
@@ -0,0 +1,241 @@
+            $parent.find('.note-editable').trigger('mouseup');
+            $parent.find('.note-editing-area').css({
+                minHeight:'100px',
+                minWidth:'260px',
+            });
+        });
+
+    },
+    set_fields_values: function(lang, tr_value) {
+        this.$('.oe_translation_field[name="' + lang +
+                '-' + this.translatable_field + '"]').val(tr_value || '').attr(
+                'data-value', tr_value || '');
+
+        this.initialize_html_fields(lang);
+    },
+    do_load_fields_values: function() {
+        var self = this,
+            deferred = [];
+
+        this.$('.oe_translation_field').val('').removeClass('touched');
+
+        var def = $.Deferred();
+        deferred.push(def);
+        rpc.query({
+            model: this.view.modelName,
+            method: 'get_field_translations',
+            args: [
+                [this.res_id],
+            ],
+            kwargs: {
+                field_name: this.translatable_field,
+            },
+        }).done(
+            function (res) {
+                if (res[self.res_id]){
+                    _.each(res[self.res_id], function(translation, lang) {
+                        self.set_fields_values(lang, translation[1]);
+                    });
+                    self.resize_textareas();
+                    def.resolve();
+                }
+            });
+
+        return deferred;
+    },
+    on_button_save: function() {
+        var translations = {},
+            self = this,
+            save_mutex = new Mutex();
+        this.$('.oe_translation_field.touched').each(function() {
+            var field = $(this).attr('name').split('-');
+            if (!translations[field[0]]) {
+                translations[field[0]] = {};
+            }
+            translations[field[0]][field[1]] = $(this).val();
+        });
+        _.each(translations, function(text, code) {
+            save_mutex.exec(function() {
+                var done = new $.Deferred(); // holds the mutex
+
+                var context = new Context(session.user_context, {lang: code});
+                rpc.query({
+                    model: self.view.modelName,
+                    method: 'write',
+                    args: [self.res_id, text],
+                    kwargs: {context: context.eval()}
+                }).then(function() {
+                    done.resolve();
+                });
+                if (code === self.view_language) {
+                    _.each(text, function(value, key) {
+                        var view_elem = self.view.$('input[name="'+ key + '"]');
+                        view_elem.val(value).trigger('change');
+                    });
+                }
+                return done;
+            });
+        });
+        this.close();
+    },
+    on_button_close: function() {
+        this.close();
+    },
+
+});
+
+FormView.include({
+    render_sidebar: function($node) {
```

Comment:
This does not seem to exist on the `FormView`, therefore not called anywhere.

Reply:
@tschanzt it's `renderSidebar` indeed. Do you have any piece of code defining `render_sidebar`?